### ISSUE TITLE
other: I can't send emails to the internet.

### ISSUE THREAD

**Issue by jpsanchezl10** [docker-mailserver/docker-mailserver#3488]
### Subject

Other

### Description

I deployed the docker with this command:

```bash
docker run -d --name=mailserver \
  --hostname="$HOSTNAME" \
  --domainname="example.com" \
  -p 25:25 -p 143:143 -p 587:587 -p 993:993 \
  -e ENABLE_SPAMASSASSIN=1 \
  -e SPAMASSASSIN_SPAM_TO_INBOX=1 \
  -e ENABLE_CLAMAV=1 \
  -e ENABLE_POSTGREY=1 \
  -e ENABLE_FAIL2BAN=0 \
  -e ENABLE_SASLAUTHD=0  \
  -e ONE_DIR=1 \
  -e TZ=America/New_York \
  -v ~/docker/mailserver/data/:/var/mail/ \
  -v ~/docker/mailserver/state/:/var/mail-state/ \
  -v ~/docker/mailserver/logs/:/var/log/mail/ \
  -v ~/docker/mailserver/config/:/tmp/docker-mailserver/ \
  --restart=unless-stopped \
  mailserver/docker-mailserver
```

I am able to receive emails from outside the internet (I set up my DNS records in Cloudflare), but I can only send email to other users inside the server (`@example.com`).

If I try to send email to Gmail, iCloud, yahoo, etc, they never receive the email.

**Comment by polarathene** [comment]
https://docker-mailserver.github.io/docker-mailserver/latest/config/debugging/#mail-sent-from-dms-does-not-arrive-at-destination

**Comment by jpsanchezl10** [comment]
Aug 19 12:00:49 webhosting postfix/smtp[751]: 160AB243C40: to=<mymail@gmail.com>, relay=none, delay=80976, delays=80941/0.02/34/0, dsn=4.4.1, status=deferred (connect to alt2.gmail-smtp-in.l.google.com[2a00:1450:400b:c00::1a]:25: Cannot assign requested address)
Aug 19 12:00:53 webhosting postfix/smtp[750]: connect to alt2.gmail-smtp-in.l.google.com[209.85.202.27]:25: No route to host
Aug 19 12:00:53 webhosting postfix/smtp[750]: 7E4BC24550C: to=<mymail@gmail.com>, relay=none, delay=87493, delays=87455/0.02/38/0, dsn=4.4.1, status=deferred (connect to alt2.gmail-smtp-in.l.google.com[209.85.202.27]:25: No route to host)
Aug 19 12:00:58 webhosting postfix/submission/smtpd[777]: connect from unknown[77.90.185.60]
Aug 19 12:01:02 webhosting postfix/smtp[746]: connect to alt1.gmail-smtp-in.l.google.com[64.233.186.27]:25: No route to host
Aug 19 12:01:02 webhosting postfix/smtp[746]: connect to alt1.gmail-smtp-in.l.google.com[2800:3f0:4003:c00::1a]:25: Cannot assign requested address
Aug 19 12:01:03 webhosting postfix/smtp[746]: connect to alt2.gmail-smtp-in.l.google.com[209.85.202.26]:25: No route to host
Aug 19 12:01:03 webhosting postfix/smtp[746]: 5CC6C243C54: to=<mymail@gmail.com>, relay=none, delay=121808, delays=121760/0.01/48/0, dsn=4.4.1, status=deferred (connect to alt2.gmail-smtp-in.l.google.com[209.85.202.26]:25: No route to host)

theese are the logs

**Comment by allddd** [comment]
This is very likely to be a firewall issue, try running:
```
telnet smtp.gmail.com 25
```
You could also use `nmap` or a similar tool to check from the outside whether ports are open at all.  
Sending an email to [mail-tester](https://www.mail-tester.com/) is also a good idea as a test, as they will accept anything no matter how badly configured the mail server is.

**Comment by jpsanchezl10** [comment]
If I run the 
certbot-cloudflare and certbot-cloudflare-renew with docker compose like this: I need to run the mailserver as normally  ?
services:
  mailserver:
  	hostname:******.com
    environments:
      # Set SSL certificate type.
      - SSL_TYPE=letsencrypt
    volumes:
      # Mount the cert folder generated by Certbot:
      - ./docker-data/certbot/certs/:/etc/letsencrypt/:ro

  certbot-cloudflare:
    image: certbot/dns-cloudflare:latest
    command: certonly --dns-cloudflare --dns-cloudflare-credentials /run/secrets/cloudflare-api-token -d ******.com
    volumes:
      - ./docker-data/certbot/certs/:/etc/letsencrypt/
      - ./docker-data/certbot/logs/:/var/log/letsencrypt/
    secrets:
      - cloudflare-api-token

  certbot-cloudflare-renew:
    image: certbot/dns-cloudflare:latest
    command: renew --dns-cloudflare --dns-cloudflare-credentials /run/secrets/cloudflare-api-token
    volumes:
      - ./docker-data/certbot/certs/:/etc/letsencrtypt/
      - ./docker-data/certbot/logs/:/var/log/letsencrypt/
    secrets:
      - cloudflare-api-token
secrets:
  cloudflare-api-token:
    file: ./docker-data/certbot/secrets/cloudflare.ini

**Comment by jpsanchezl10** [comment]
> This is very likely to be a firewall issue, try running:
> 
> ```shell
> telnet smtp.gmail.com 25
> ```
> 
> Also try telnet-ing your server from outside:
> 
> ```shell
> telnet mail.example.com 25
> ```
> 
> and watch the log to see if anything happens:
> 
> ```shell
> docker logs -f container-name
> ```
> 
> You could also use `nmap` or a similar tool to check from the outside whether ports are open at all. Sending an email to [mail-tester](https://www.mail-tester.com/) is also a good idea as a test, as they will accept anything no matter how badly configured the mail server is.

im getting theese logs ( my telnet succeded )
Aug 19 13:27:27 webhosting postfix/submission/smtpd[7724]: connect from unknown[77.90.185.60]

Aug 19 13:27:31 webhosting dovecot: auth: passwd-file(shop@relu.mx,77.90.185.60): unknown user (SHA1 of given password: 971a8a)

Aug 19 13:27:33 webhosting postfix/submission/smtpd[7724]: warning: unknown[77.90.185.60]: SASL LOGIN authentication failed: UGFzc3dvcmQ6

Aug 19 13:27:34 webhosting postfix/submission/smtpd[7724]: disconnect from unknown[77.90.185.60] ehlo=1 auth=0/1 rset=1 quit=1 commands=3/4

Aug 19 13:27:57 webhosting postfix/submission/smtpd[7724]: connect from unknown[77.90.185.60]

Aug 19 13:28:00 webhosting dovecot: auth: passwd-file(shopping@relu.mx,77.90.185.60): unknown user (SHA1 of given password: 971a8a)
this means someone is tying to auth to my server ?

**Comment by polarathene** [comment]
> These are the logs:

You misunderstood. I linked you to a section of the docs about sending mail. It is a common problem that when you rent a server, or use many ISP that the outbound port 25 is blocked. This prevents you from sending mail, while the inbound port 25 may be open allowing you to receive mail.

This is to prevent spammers. The docs section I linked you to provides links to Vultr, AWS, and Azure pages with this information. Decent services that block this port will have information about that and advice on how to unblock it if they allow it.

If you cannot find out that information, or cannot get the port unblocked, then you will want to relay your mail through a service like [SendGrid](https://sendgrid.com/). We have some [docs on configuring DMS to send mail to a relay service](https://docker-mailserver.github.io/docker-mailserver/latest/config/advanced/mail-forwarding/relay-hosts/) like SendGrid, you can try this and should be successful when sending mail over a different port that is not blocked (_possibly 587, 465, or relay service specific ports offered, sometimes port 2525_).

**Comment by polarathene** [comment]
@jpsanchezl10 can you confirm if the problem was port 25 outbound was blocked?

It's helpful to share why you close an issue when someone comes across the issue from a search result :)

**Comment by jpsanchezl10** [comment]
I closed by accident , the problem persists.
I set the SMTP port to 587 with the env variable (SMTP_PORT=587) and exposed the port ("587:587")
and the server still used the port 25 for no reason :

`Aug 20 01:47:50 webhosting postfix/smtp[782]: connect to gmail-smtp-in.l.google.com[142.251.116.27]:25: No route to host`

I also tried to set the out port like this (465:25) so the container will send thought 25 to my machine and my machine forward with 465, but I have the same issue.

**Comment by jpsanchezl10** [comment]
I set up a relay with sendgrid and it is not working also not working :
this is my docker-compose.yaml:

```
#final focker compose:
version: '3.8'

services:
  mailserver:
    image: mailserver/docker-mailserver:latest
    container_name: mailserver
    hostname: webhosting
    domainname: mail.MYDOMAIN.MX
    ports:
      - "25:25"
      - "143:143"
      - "587:587"
      - "993:993"
      - "465:465"
    environment:
      - ENABLE_SPAMASSASSIN=1
      - SPAMASSASSIN_SPAM_TO_INBOX=1
      - ENABLE_CLAMAV=1
      - ENABLE_POSTGREY=1
      - ENABLE_FAIL2BAN=1
      - ONE_DIR=1
      - TZ=America/New_York
      - ENABLE_SASLAUTHD=0
      - RELAY_HOST=smtp.sendgrid.net
      - RELAY_PORT=25
      - RELAY_USER=apikey
      - RELAY_PASSWORD=<MY_API_KEY>
      #- SSL_TYPE=letsencrypt

    volumes:
      - ~/docker/mailserver/data/:/var/mail/
      - ~/docker/mailserver/state/:/var/mail-state/
      - ~/docker/mailserver/logs/:/var/log/mail/
      - ~/docker/mailserver/config/:/tmp/docker-mailserver/
      - ./docker-data/certbot/certs/:/etc/letsencrypt/:ro
    restart: always


  certbot-cloudflare:
    image: certbot/dns-cloudflare:latest
    command: certonly --dns-cloudflare --dns-cloudflare-credentials /run/secrets/cloudflare-api-token -d mail.relu.mx
    volumes:
      - ./docker-data/certbot/certs/:/etc/letsencrypt/
      - ./docker-data/certbot/logs/:/var/log/letsencrypt/
    secrets:
      - cloudflare-api-token

  certbot-cloudflare-renew:
    image: certbot/dns-cloudflare:latest
    command: renew --dns-cloudflare --dns-cloudflare-credentials /run/secrets/cloudflare-api-token
    volumes:
      - ./docker-data/certbot/certs/:/etc/letsencrtypt/
      - ./docker-data/certbot/logs/:/var/log/letsencrypt/
    secrets:
      - cloudflare-api-token

secrets:
  cloudflare-api-token:
    file: ./docker-data/certbot/secrets/cloudflare.ini

```
Logs:
```
Aug 20 03:01:03 webhosting postfix/smtp-amavis/smtp[1740]: 087E9245598: to=<myprivatemail@gmail.com>, relay=127.0.0.1[127.0.0.1]:10024, delay=0.56, delays=0.05/0/0/0.5, dsn=2.0.0, status=sent (250 2.0.0 from MTA(smtp:[127.0.0.1]:10025): 250 2.0.0 Ok: queued as 8F674245599)

Aug 20 03:01:03 webhosting postfix/qmgr[699]: 087E9245598: removed

Aug 20 03:01:03 webhosting postfix/error[1748]: 8F674245599: to=<myprivateemail@gmail.com>, relay=none, delay=0.01, delays=0/0/0/0, dsn=4.4.1, status=deferred (delivery temporarily suspended: connect to smtp.sendgrid.net[167.89.118.58]:25: No route to host)

Aug 20 03:01:03 webhosting dovecot: imap-login: Login: user=<admin@mydomain.mx>, method=PLAIN, rip=172.18.0.1, lip=172.18.0.2, mpid=1754, session=<qfg9U1UDaL+sEgAB>

Aug 20 03:01:03 webhosting dovecot: imap(admin@mydomain.mx)<1754><qfg9U1UDaL+sEgAB>: Disconnected: Logged out in=70 out=709 deleted=0 expunged=0 trashed=0 hdr_count=0 hdr_bytes=0 body_count=0 body_bytes=0

Aug 20 03:01:36 webhosting postfix/error[1748]: 45F2924558F: to=<myprivateemail@gmail.com>, relay=none, delay=562, delays=562/0/0/0, dsn=4.4.1, status=deferred (delivery temporarily suspended: connect to smtp.sendgrid.net[167.89.118.58]:25: No route to host)

Aug 20 03:01:36 webhosting postfix/error[1748]: 45F2924558F: to=<myprivateemail@icloud.com>, relay=none, delay=562, delays=562/0/0/0, dsn=4.4.1, status=deferred (delivery temporarily suspended: connect to smtp.sendgrid.net[167.89.118.58]:25: No route to host)

Aug 20 03:01:36 webhosting postfix/error[1748]: 45F2924558F: to=<test-2b9l3y3ss@srv1.mail-tester.com>, relay=none, delay=562, delays=562/0/0/0, dsn=4.4.1, status=deferred (delivery temporarily suspended: connect to smtp.sendgrid.net[167.89.118.58]:25: No route to host)

```

**Comment by jpsanchezl10** [comment]
**I also set up a relay with sendgrid.com and the issue persists:**
docker-compose.yaml:
```
#final focker compose:
version: '3.8'

services:
  mailserver:
    image: mailserver/docker-mailserver:latest
    container_name: mailserver
    hostname: webhosting
    domainname: mail.MYDOMAIN.MX
    ports:
      - "25:25"
      - "143:143"
      - "587:587"
      - "993:993"
      - "465:465"
    environment:
      - ENABLE_SPAMASSASSIN=1
      - SPAMASSASSIN_SPAM_TO_INBOX=1
      - ENABLE_CLAMAV=1
      - ENABLE_POSTGREY=1
      - ENABLE_FAIL2BAN=1
      - ONE_DIR=1
      - TZ=America/New_York
      - ENABLE_SASLAUTHD=0
      - RELAY_HOST=smtp.sendgrid.net
      - RELAY_PORT=25
      - RELAY_USER=apikey
      - RELAY_PASSWORD=<MY_API_KEY>
      #- SSL_TYPE=letsencrypt

    volumes:
      - ~/docker/mailserver/data/:/var/mail/
      - ~/docker/mailserver/state/:/var/mail-state/
      - ~/docker/mailserver/logs/:/var/log/mail/
      - ~/docker/mailserver/config/:/tmp/docker-mailserver/
      - ./docker-data/certbot/certs/:/etc/letsencrypt/:ro
    restart: always


  certbot-cloudflare:
    image: certbot/dns-cloudflare:latest
    command: certonly --dns-cloudflare --dns-cloudflare-credentials /run/secrets/cloudflare-api-token -d mail.relu.mx
    volumes:
      - ./docker-data/certbot/certs/:/etc/letsencrypt/
      - ./docker-data/certbot/logs/:/var/log/letsencrypt/
    secrets:
      - cloudflare-api-token

  certbot-cloudflare-renew:
    image: certbot/dns-cloudflare:latest
    command: renew --dns-cloudflare --dns-cloudflare-credentials /run/secrets/cloudflare-api-token
    volumes:
      - ./docker-data/certbot/certs/:/etc/letsencrtypt/
      - ./docker-data/certbot/logs/:/var/log/letsencrypt/
    secrets:
      - cloudflare-api-token

secrets:
  cloudflare-api-token:
    file: ./docker-data/certbot/secrets/cloudflare.ini

```
**Docker Logs:**
```
Aug 20 03:01:03 webhosting postfix/smtp-amavis/smtp[1740]: 087E9245598: to=<myprivatemail@gmail.com>, relay=127.0.0.1[127.0.0.1]:10024, delay=0.56, delays=0.05/0/0/0.5, dsn=2.0.0, status=sent (250 2.0.0 from MTA(smtp:[127.0.0.1]:10025): 250 2.0.0 Ok: queued as 8F674245599)

Aug 20 03:01:03 webhosting postfix/qmgr[699]: 087E9245598: removed

Aug 20 03:01:03 webhosting postfix/error[1748]: 8F674245599: to=<myprivateemail@gmail.com>, relay=none, delay=0.01, delays=0/0/0/0, dsn=4.4.1, status=deferred (delivery temporarily suspended: connect to smtp.sendgrid.net[167.89.118.58]:25: No route to host)

Aug 20 03:01:03 webhosting dovecot: imap-login: Login: user=<admin@mydomain.mx>, method=PLAIN, rip=172.18.0.1, lip=172.18.0.2, mpid=1754, session=<qfg9U1UDaL+sEgAB>

Aug 20 03:01:03 webhosting dovecot: imap(admin@mydomain.mx)<1754><qfg9U1UDaL+sEgAB>: Disconnected: Logged out in=70 out=709 deleted=0 expunged=0 trashed=0 hdr_count=0 hdr_bytes=0 body_count=0 body_bytes=0

Aug 20 03:01:36 webhosting postfix/error[1748]: 45F2924558F: to=<myprivateemail@gmail.com>, relay=none, delay=562, delays=562/0/0/0, dsn=4.4.1, status=deferred (delivery temporarily suspended: connect to smtp.sendgrid.net[167.89.118.58]:25: No route to host)

Aug 20 03:01:36 webhosting postfix/error[1748]: 45F2924558F: to=<myprivateemail@icloud.com>, relay=none, delay=562, delays=562/0/0/0, dsn=4.4.1, status=deferred (delivery temporarily suspended: connect to smtp.sendgrid.net[167.89.118.58]:25: No route to host)

Aug 20 03:01:36 webhosting postfix/error[1748]: 45F2924558F: to=<test-2b9l3y3ss@srv1.mail-tester.com>, relay=none, delay=562, delays=562/0/0/0, dsn=4.4.1, status=deferred (delivery temporarily suspended: connect to smtp.sendgrid.net[167.89.118.58]:25: No route to host)

```

**Comment by polarathene** [comment]
> I set the SMTP port to 587 with the env variable (SMTP_PORT=587) and exposed the port ("587:587") and the server still used the port 25 for no reason

You seem to misunderstand:
- There is no `SMTP_PORT` ENV, perhaps you meant `RELAY_PORT`?
- Traffic **can enter** your server on port 25. (mail delivered to DMS)
- Traffic **cannot exit** your server on port 25 (DMS sending mail out); This is because your server is blocking it to prevent abuse by spammers.

`RELAY_PORT=25` won't do you any good, even with SendGrid because of this. You need to use another port SendGrid allows, it should be 587, 465 or similar in their documentation (or you can just try them).

After this is done, your relay username (`RELAY_USER`) and password (`RELAY_PASSWORD`) will be used to login to SendGrid server (`RELAY_HOST`) and send the mail to SendGrid to handle, which will then send out your mail (_on their port 25_) to the target recipient on your behalf.

**Comment by polarathene** [comment]
Also this part of your configuration is definitely not right:

```yaml
hostname: webhosting
domainname: mail.MYDOMAIN.MX
```

You should only use `hostname, no `domainname`.
- This has nothing to do with email address you want to use like `hello@example.com` that you add with `setup email add hello@example.com`.
- It is for your mail server to talk to other mail software. Usually you give it the subdomain `mail` and update DNS records to refer to it.

Usually you would use `hostname: mail.example.com`. Here are some references that may be helpful if you still find connection problems with network, make sure DNS setup correctly:
- https://github.com/docker-mailserver/docker-mailserver/pull/3465#discussion_r1294061262
- https://github.com/docker-mailserver/docker-mailserver/issues/3462#issuecomment-1669129008
- https://docker-mailserver.github.io/docker-mailserver/latest/usage/#advanced-dns-setup-dkim-dmarc-spf

If you still have error about routing, this is something you will need to resolve on your end as it may be a network issue with your server provider, not something related to DMS.

**Comment by jpsanchezl10** [comment]
It worked changing the port to 587 in the relay , than you very much .
